### PR TITLE
feat: Use AbstractConfig for API server config

### PR DESCRIPTION
**Pull request by purplefox**
### Description 

Issue: https://github.com/confluentinc/ksql/issues/4330

This PR introduces a new config based on AbstractConfig for configuring the API server.

The current config param names and docs are kind of provisional as they're likely to change a fair bit as the server is integrated into the ksqlDB back end, so don't read too much into them right now.

### Testing done 

API tests pass

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import io.confluent.ksql.util.KsqlConfig;
+import io.vertx.core.json.JsonObject;
+import java.util.Map;
+
+/**
+ * Config for the API server
+ */
+public class ApiServerConfig extends AbstractConfig {
+
+  private static final String PROPERTY_PREFIX = "apiserver.";
```
Comment (big-andy-coates) [inline ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java]: Can we add  a `ksql.` prefix to this please?

File: ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  public ApiServerConfig(final Map<?, ?> map) {
+    super(CONFIG_DEF, map);
+  }
+
+  public ApiServerConfig(final JsonObject jsonObject) {
+    super(CONFIG_DEF, jsonObject.getMap());
+  }
+
+  public JsonObject toJsonObject() {
+    return new JsonObject(originalsWithPrefix(""));
```
Comment (big-andy-coates) [inline ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java]: I think using `originalsWithPrefix` will mean the returned JsonObject won't pick up any defaults defined in the config.   A unit test would check this.

File: ksql-api/src/main/java/io/confluent/ksql/api/server/Server.java [hunk 3]
```diff
@@ -36,48 +36,56 @@
...
     this.endpoints = Objects.requireNonNull(endpoints);
-    this.httpServerOptions = Objects.requireNonNull(httpServerOptions);
+  }
+
+  private HttpServerOptions createHttpServerOptions(final ApiServerConfig apiServerConfig) {
+    return
+        new HttpServerOptions().setHost(apiServerConfig.getString(ApiServerConfig.LISTEN_HOST))
+            .setPort(apiServerConfig.getInt(ApiServerConfig.LISTEN_PORT))
+            .setUseAlpn(true)
+            .setSsl(true)
```
Comment (big-andy-coates) [inline ksql-api/src/main/java/io/confluent/ksql/api/server/Server.java]: I would of thought we'd want ssl to be optional... could drive it off the presence or absence of `ApiServerConfig.KEY_PATH`?

File: ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+          Type.STRING,
+          null,
+          Importance.MEDIUM,
+          KEY_PATH_DOC)
+      .define(
+          CERT_PATH,
+          Type.STRING,
+          null,
+          Importance.MEDIUM,
+          CERT_PATH_DOC);
```
Comment (big-andy-coates) [inline ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java]: FYI, you can set `Validator`s on these, e.g. to fail early if the port is negative etc.

While on the subject of ports - Jetty supports auto-port assignment if port is zero. Does vert.x have something similar?  I've run into bind exceptions locally when doing multiple things - build servers could run into the same issue...

File: ksql-api/src/main/java/io/confluent/ksql/api/server/Server.java (hunk 3 above)
Comment (purplefox) [reply ksql-api/src/main/java/io/confluent/ksql/api/server/Server.java]: Yes, we could. But HTTP2 is usually always used with TLS - all major browsers require TLS with HTTP2. I know we're not aiming at the browser case but perhaps we should follow suit?

File: ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java (hunk 4 above)
Comment (purplefox) [reply ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java]: Yes, this is supported by Vert.x https://vertx.io/docs/apidocs/io/vertx/core/http/HttpServer.html#actualPort--

File: ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java (hunk 1 above)
Comment (purplefox) [reply ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java]: It's a little bit further down in the file :)

File: ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java (hunk 2 above)
Comment (purplefox) [reply ksql-api/src/main/java/io/confluent/ksql/api/server/ApiServerConfig.java]: It's ok. Checked with unit test.

**Pull request merged**